tests/test_package_client.py

```diff
@@ -113,56 +98,42 @@ def test_start_client(self, setup_tester_ssh_connection):
         assert "State transition: authorize [Sync] -> authorize-wait [Idle]" in result.stdout
 
     @pytest.mark.usefixtures("setup_test_container")
-    def test_stop_client(self, setup_tester_ssh_connection):
-        setup_tester_ssh_connection.run('sudo systemctl stop mender')
-        time.sleep(1)
-        result = setup_tester_ssh_connection.run('sudo journalctl -u mender --no-pager')
-
-        # Check authorization failure
-        assert "Stopping Mender OTA update service..." in result.stdout
-        assert "Stopped Mender OTA update service." in result.stdout
-
-    @pytest.mark.usefixtures("setup_test_container")
-    def test_remove_package(self, setup_tester_ssh_connection, mender_dist_packages_versions):
+    def test_remove_stop(self, setup_tester_ssh_connection, mender_dist_packages_versions):
         result = setup_tester_ssh_connection.run('sudo dpkg -r mender-client')
         assert "Removing mender-client (" + mender_dist_packages_versions["mender-client"] + ")" in result.stdout
 
-        # Check directories
+        # Check files and directories
         setup_tester_ssh_connection.run('test ! -f /usr/bin/mender')
         setup_tester_ssh_connection.run('test ! -e /usr/share/mender')
+        setup_tester_ssh_connection.run('test ! -f /lib/systemd/system/mender-client.service')
+        setup_tester_ssh_connection.run('test ! -f /etc/systemd/system/multi-user.target.wants/mender-client.service')
         setup_tester_ssh_connection.run('test -d /etc/mender')
+        setup_tester_ssh_connection.run('test -f /etc/mender/mender.conf')
         setup_tester_ssh_connection.run('test -d /var/lib/mender/')
+        setup_tester_ssh_connection.run('test -f /var/lib/mender/device_type')
+
+        result = setup_tester_ssh_connection.run('sudo journalctl -u mender-client --no-pager')
+        assert "Stopping Mender OTA update service..." in result.stdout
+        assert "Stopped Mender OTA update service." in result.stdout
```

Comment:
Would it make sense with a negative `pgrep mender` check here as well?

Reply:
Good idea, I will add it.